src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -96,6 +109,26 @@ def register_options(cls, register):
                 "treated as a potential dependency if this option is set to 2 but not if set to 3."
             ),
         )
+        register(
+            "--assets",
```

Comment:
This makes me want "sub" subsystems.
E.g. `python-infer.assets` `python-infer.imports`, etc...